# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/udev/hotfix-arch.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- systemd-257.7/src/basic/architecture.h.vanilla	2025-07-24 15:26:38.036804635 +0200
+++ systemd-257.7/src/basic/architecture.h	2025-07-24 15:29:56.183603974 +0200
@@ -26,6 +26,7 @@
         ARCHITECTURE_MIPS64_LE,
         ARCHITECTURE_MIPS_LE,
         ARCHITECTURE_NIOS2,
+        ARCHITECTURE_OR1K,
         ARCHITECTURE_PARISC,
         ARCHITECTURE_PARISC64,
         ARCHITECTURE_PPC,
@@ -220,6 +221,10 @@
 #elif defined(__nios2__)
 #  define native_architecture() ARCHITECTURE_NIOS2
 #  define LIB_ARCH_TUPLE "nios2-linux-gnu"
+#elif defined(__or1k__)
+#  define native_architecture() ARCHITECTURE_OR1K
+#  define LIB_ARCH_TUPLE "or1k-linux-gnu"
+#
 #elif defined(__riscv)
 #  if __SIZEOF_POINTER__ == 4
 #    define native_architecture() ARCHITECTURE_RISCV32
--- systemd-257.7/src/basic/architecture.c.vanilla	2025-07-24 15:25:24.525017109 +0200
+++ systemd-257.7/src/basic/architecture.c	2025-07-24 15:28:33.781400055 +0200
@@ -83,6 +83,9 @@
 #elif defined(__nios2__)
                 { "nios2",      ARCHITECTURE_NIOS2    },
 
+#elif defined(__or1k__)
+                { "openrisc",   ARCHITECTURE_OR1K     },
+
 #elif defined(__powerpc__) || defined(__powerpc64__)
                 { "ppc64le",    ARCHITECTURE_PPC64_LE },
                 { "ppc64",      ARCHITECTURE_PPC64    },
@@ -158,6 +161,7 @@
         [ARCHITECTURE_MIPS_LE]     = "mips-le",
         [ARCHITECTURE_MIPS]        = "mips",
         [ARCHITECTURE_NIOS2]       = "nios2",
+        [ARCHITECTURE_OR1K]        = "openrisc",
         [ARCHITECTURE_PARISC64]    = "parisc64",
         [ARCHITECTURE_PARISC]      = "parisc",
         [ARCHITECTURE_PPC64_LE]    = "ppc64-le",
